whois-api/src/main/java/net/ripe/db/whois/api/apiKey/BearerTokenExtractor.java

```diff
@@ -60,40 +67,40 @@ private OAuthSession getOAuthSession(final String bearerToken, final String apiK
         }
 
         try {
-            final SignedJWT signedJWT = SignedJWT.parse(StringUtils.substringAfter(bearerToken, "Bearer "));
-
-            if(!verifyJWTSignature(signedJWT)) {
-              LOGGER.debug("JWT signature verification failed for {}", apiKeyId);
-              return new OAuthSession(apiKeyId);
-            }
-
-            //TODO[MA]: remove when apiKeyId is available from api registry call
-            return OAuthSession.from(new ObjectMapper().readValue(signedJWT.getPayload().toString(), OAuthSession.class), apiKeyId);
-        } catch (JsonProcessingException e) {
-            LOGGER.error("Failed to serialize OAuthSession, this should never have happened", e);
-            return  new OAuthSession(apiKeyId);
-        } catch (Exception e) {
-            LOGGER.error("Failed to read OAuthSession, this should never have happened", e);
-            return new OAuthSession(apiKeyId);
-        }
-    }
+            final String accessToken = StringUtils.substringAfter(bearerToken, "Bearer ");
+            final SignedJWT signedJWT = SignedJWT.parse(accessToken);
 
-    private boolean verifyJWTSignature(final SignedJWT signedJWT) {
-        try {
-            final List<RSAKey> rsaKeys = apiPublicKeyLoader.loadPublicKey();
+            final ConfigurableJWTProcessor<SecurityContext> jwtProcessor = new DefaultJWTProcessor<>();
 
-            if(rsaKeys.isEmpty()) {
-              LOGGER.debug("RSA public key is not available");
-              return true;
-            }
+            final JWKSource<SecurityContext> keySource = JWKSourceBuilder
+                    .create(new URL(jwksSetUrl))
```

Comment:
I think that URL is deprecated. I remember to have this issue with NRTM Client
You can use`URI.create(jwksSetUrl).toURL()`